src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -23,35 +24,45 @@
 import com.google.api.codegen.util.go.GoTypeTable;
 import com.google.api.codegen.util.Name;
 import com.google.api.tools.framework.aspects.documentation.model.DocumentationUtil;
+import com.google.api.tools.framework.model.Interface;
 import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
 import com.google.api.tools.framework.model.ProtoElement;
 import com.google.api.tools.framework.model.TypeRef;
 
+import io.grpc.Status;
+
+import java.io.File;
 import java.util.List;
 
 public class GoSurfaceNamer extends SurfaceNamer {
-  private final String serviceName;
+  private final Model model;
+  private final Interface service;
+  private final ApiConfig apiConfig;
   private final GoModelTypeNameConverter converter;
 
-  public GoSurfaceNamer(String serviceName) {
-    this(serviceName, new GoModelTypeNameConverter());
+  public GoSurfaceNamer(Model model, Interface service, ApiConfig apiConfig) {
```

Comment:
The Model/Interface/ApiConfig objects should be passed into functions in this class, not the constructor, because the cardinality of SurfaceNamer and those classes isn't always the same. For example, see here:

https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java#L95

In that circumstance, there are multiple interfaces to mock, but one SurfaceNamer.